src/tdlib/chat_history.rs

```diff
@@ -3,12 +3,33 @@ use gtk::subclass::prelude::*;
 use gtk::{gio, glib};
 use std::cmp::Ordering;
 use std::collections::hash_map::Entry;
+use std::error::Error;
+use std::fmt::Display;
 use tdlib::enums::{self, Update};
 use tdlib::functions;
 use tdlib::types::Message as TelegramMessage;
 
 use crate::tdlib::{Chat, ChatHistoryItem, ChatHistoryItemType, Message};
 
+#[derive(Debug)]
+pub(crate) enum ChatHistoryError {
+    AlreadyLoading,
+    Tdlib(tdlib::types::Error),
+}
+
+impl Display for ChatHistoryError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ChatHistoryError::AlreadyLoading => {
+                write!(f, "The chat history is already loading messages")
+            }
+            ChatHistoryError::Tdlib(td_error) => write!(f, "TDLib error: {:?}", td_error),
+        }
+    }
+}
+
+impl Error for ChatHistoryError {}
+
```

Comment:
I suggest that we use the [thiserror](https://crates.io/crates/thiserror) crate for defining custom error types. This will reduce the boilerplate a lot